packages/block-library/src/template-part/index.php

```diff
@@ -126,27 +126,39 @@ function render_block_core_template_part( $attributes ) {
 			'';
 	}
 
+	// Look up area definition.
+	$area_definition = null;
+	$defined_areas   = get_allowed_block_template_part_areas();
+	foreach ( $defined_areas as $defined_area ) {
+		if ( $defined_area['area'] === $area ) {
+			$area_definition = $defined_area;
+			break;
+		}
+	}
+
+	// If $area is not allowed, set it back to the uncategorized default.
+	if ( ! $area_definition ) {
+		$area = WP_TEMPLATE_PART_AREA_UNCATEGORIZED;
+	}
+
 	// Run through the actions that are typically taken on the_content.
 	$seen_ids[ $template_part_id ] = true;
 	$content                       = do_blocks( $content );
 	unset( $seen_ids[ $template_part_id ] );
 	$content = wptexturize( $content );
 	$content = convert_smilies( $content );
 	$content = shortcode_unautop( $content );
-	$content = wp_filter_content_tags( $content );
+	$content = wp_filter_content_tags( $content, "the_template_part_{$area}" );
```

Comment:
Why use the prefix `the_`? Introducing historical conventions from b2/cafelog into block themes seems very backward. Some care should be taken to choose the right string, as it probably can't ever be changed.

Reply:
@carlomanf I agree with you on the one hand, one the other hand the values used as `$context` in core so far are those older ones based on filters, e.g. `the_content`, `the_post_thumbnail`. I chose this string to align with that.

But if you feel strongly about getting rid of the `the_` prefix despite other context values using it, I'm certainly open to doing that.

Reply:
Not all of the context values that currently pass through `wp_filter_content_tags` start with `the_`, only the ones based on b2-style template tags. `widget_text_content` and `widget_block_content` also pass through `wp_filter_content_tags` so I think there is full freedom to choose a name that describes the context well, i.e. that it's a block callback.

You may want to look at WordPress/wordpress-develop#3851 and whether the function call there should use the same value.

Reply:
@carlomanf Fair point, will update. 👍